js/fileupload.js

```diff
@@ -353,13 +353,50 @@ var insertImageInTinyMCE = function(editor, image) {
  */
 if (typeof tinyMCE != 'undefined') {
     tinyMCE.PluginManager.add('glpi_upload_doc', function(editor) {
+        let last_paste_content = null;
+        const rtf_img_types = {
+            'pngblip': 'image/png',
+            'jpegblip': 'image/jpeg',
+        };
+        editor.on('paste', (e) => {
+            last_paste_content = e.clipboardData;
+        });
         editor.on('PastePreProcess', function(event) {
+            const base64_img_contents = [];
+            if (last_paste_content !== null && last_paste_content.types.includes('text/rtf')) {
+                // Extract all RTF images and remove line breaks
+                const rtf_content = last_paste_content.getData('text/rtf');
+                const rtf_content_no_line_break = rtf_content.replace(/(\r\n|\n|\r)/gm, "");
+                const hex_binary = rtf_content_no_line_break.matchAll(/\\(pngblip|jpegblip)([a-z0-9]*)}/g);
+
+                // For each match, convert to base64
+                for (const match of hex_binary) {
+                    const img_type = match[1];
+                    const hex = match[2];
+                    const hexToBase64 = function(hexstring) {
+                        return btoa(hexstring.match(/\w{2}/g).map(function(a) {
+                            return String.fromCharCode(parseInt(a, 16));
+                        }).join(""));
+                    };
+                    base64_img_contents.push({
+                        type: rtf_img_types[img_type],
+                        content: hexToBase64(hex)
+                    });
+                }
+            }
             // Trigger upload process for each pasted image
             var fragment = $('<div></div>');
             fragment.append(event.content);
             fragment.find('img').each(function() {
                 const image = $(this);
-                const src = image.attr('src');
+                let src = image.attr('src');
+                const file_pattern = '^file://';
+
+                if (src.match(file_pattern) !== null) {
```

Comment:
To handle any tool that would not provide corresponding RTF content.
```suggestion
                if (src.match(file_pattern) !== null && base64_img_contents.length > 0) {
```